morpheus/_lib/src/stages/triton_inference.cpp

```diff
@@ -113,7 +113,7 @@ InferenceClientStage::subscribe_fn_t InferenceClientStage::build_operator()
                 // When our tensor lengths are longer than our dataframe we will need to use the seq_ids
                 // array to lookup how the values should map back into the dataframe
                 const bool needs_seq_ids = x->mess_count != x->count;
-                auto reponse_memory      = std::make_shared<ResponseMemory>(x->mess_count);
+                ResponseMemory reponse_memory(x->mess_count);
```

Comment:
Creating a value type of `ResponseMemory` just to move it into a `shared_ptr<ResponseMemoryProbs>` is awkward. Instead,

1. Dont create a `ReponseMemory` object at all. Just create a `std::map<std::string, TensorObject>` and store all of the tensors in there.
2. Create a new constructor for `ResponseMemoryProbs` that accepts a `tensor_map_t` object
   3. Similar to the move constructor overload in `ResponseMemoryProbs`, check for the tensor named `"probs"`
4. After the for loop that creates all of the `TensorObject`, create a `ResponseMemoryProbs` using the new constructor overload

Reply:
Agreed that makes it much better